streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -601,6 +608,10 @@ boolean runLoop() {
                 if (size != -1L) {
                     cacheResizer.accept(size);
                 }
+                final long bufferBytesSize = maxBufferResizeSize.getAndSet(-1L);
+                if (size != -1) {
```

Comment:
This does not look right to me: why we use the size value read from `cacheResizeSize` to assign to `maxBufferSizeBytes`? They should be totally orthogonal.